src/main/java/com/softserve/academy/spaced/repetition/service/validators/PasswordMatchesValidator.java

```diff
@@ -21,17 +23,20 @@
 
     @Override
     public void initialize(PasswordMatches constraintAnnotation) {
+        /**
+         * Initializes the validator in preparation for isValid(Object, ConstraintValidatorContext) calls.
+         * This method is guaranteed to be called before any use of this instance for validation.
+         * @param constraintAnnotation
+         */
     }
 
     @Override
     public boolean isValid(String password, ConstraintValidatorContext context) {
         boolean valid = false;
         try {
-            User user = userService.getAuthorizedUser();
-            Account account = user.getAccount();
-            String pass = account.getPassword();
-            valid = passwordEncoder.matches(password, pass);
-        } catch (NotAuthorisedUserException ignored) {
+            valid = passwordEncoder.matches(password, userService.getAuthorizedUser().getAccount().getPassword());
+        } catch (NotAuthorisedUserException ex) {
+            LOGGER.error("Operation is unavailable for unauthorized users!");
```

Comment:
There are two problems here:

1. There is no unhandled error here (validation is to prevent it), but you are logging at ERROR level which should be used to signal that there is an error.
2. Sonar will ask either to log or to throw again the error.

Here is what I write in such cases.

```
LOGGER.trace("Ignored exception.", ex);
LOGGER.warn("Operation is unavailable for unauthorized users!");
```

Reply:
@antonbakalets , when sonar will ask this? I checked all code and there was no issue.

Reply:
Strange, possibly it is a rule on my project. Ok, but if it is reported on sonarcloud.io you will have to fix it.

Reply:
It can be reported after merge?

Reply:
SonarLint plugin and sonarcloud can have a litle bit different rules defined. By the way, the plugin can be configured to sync with server, but that is another story.